boxtree/tools.py

```diff
@@ -822,16 +823,20 @@ def done(self):
 
 # {{{ MPI launcher
 
-def run_mpi(script, num_processes, env):
+def run_mpi(script: str, num_processes: int, env: Dict[str, Any]) -> None:
     """Launch MPI processes.
 
-    This function forks another process and uses mpiexec to launch *num_processes*
-    copies of program *script*.
+    This function forks another process and uses ``mpiexec`` to launch
+    *num_processes* MPI processes running *script*.
 
-    :arg script: the Python script to run
-    :arg num_processes: the number of copies of program to launch
-    :arg env: a Python `dict` of environment variables
+    :arg script: the Python script to run.
+    :arg num_processes: the number of MPI process to launch.
+    :arg env: a :class:`dict` of environment variables.
     """
+    import os
+    env = {key: str(value) for key, value in env.items()}
+    env = {**os.environ, **env}
```

Comment:
This is unrelated, but it felt nicer to stringify and copy the `environ` here than make the caller do it.